docs/source/api_reference/pydax.schema.rst

```diff
@@ -0,0 +1,58 @@
+pydax.schema
```

Comment:
Do you have to explicitly write out all modules like this? Looks like you shouldn't need to do so according to the following two answers:

https://stackoverflow.com/a/21665947/1150462

https://stackoverflow.com/a/62613202/1150462

Also it doesn't seem like you have to call `sphinx-autogen` separately. Let me know how it goes

Reply:
> Also it doesn't seem like you have to call sphinx-autogen separately. Let me know how it goes

Yes it seems you can set `autosummary_generate` in `conf.py` which I'll do. 

> Do you have to explicitly write out all modules like this?

I spent some time after reading through the stackoverflow answers trying to get the custom templates working. Unfortunately I don't think those solutions will work in our case because of how we set up our source code to be hidden in private modules. The solutions rely on `autosummary` exposing variables like `classes` and `functions` which according to the [documentation](https://documentation.help/Sphinx/autosummary.html#customizing-templates):

```
functions
List containing names of “public” functions in the module. Here, “public” here means that the name does not start with an underscore. Only available for modules.

classes
List containing names of “public” classes in the module. Only available for modules.
```

However after testing this out, these variables don't seem to take into account what's defined in `__all__`, they only extract the source code defined in the module. Also Jinja templating also doesn't seem to allow me to extract the contents of `__all__` when I try to call it on a module. 

Do you have any ideas to get around these restrictions? I couldn't find anyone with the same issue online, so maybe I can post an issue in the sphinx repo. 

The `:members:` option in `autodoc` does work with `autoclass` so I can use a template like PyTorch [does](https://github.com/pytorch/pytorch/blob/master/docs/source/_templates/classtemplate.rst) do reduce having to write out all the methods/attributes a class has, but then I don't think we'll have stub files for these.

Reply:
My understanding is that if we want the document of `load_schemata()` etc. to show up in the public module instead of the private one, we have to manually write them out, is it right? I don't know at this moment how to get around those, likely I have to play with it for a while before I know the answer. If we have to manually specify, we can do that (like what sphinx in the past used to require).

Having everything in doc automated is not a priority though, we just need to have decent automation for now. We can open an issue to address these issues later so this PR doesn't get inflated too much.

Reply:
> My understanding is that if we want the document of load_schemata() etc. to show up in the public module instead of the private one, we have to manually write them out, is it right?

Yeah exactly. Since `autosummary` doesn't seem to track private modules in the `modules` variable, you would have to specify which private modules you want documented, and then the automation solutions from the stackoverflow answers work, but of course in that case the members are all documented as coming from the private modules.

Reply:
Do you prefer the stub file approach where every method/attribute gets its open page, or should we have classes fully documented on the same page? The latter is easily automated using the `:members:` option. For the former I think we'd still have to manually write out all the members.

Reply:
I think either works for now. If the library grows really big, we would have to use the first one. Since this change is unlikely an overhaul, either should work for now.